diff --git a/src/Makefile.housekeeping b/src/Makefile.housekeeping
index cac6efcf..6115bdff 100644
--- a/src/Makefile.housekeeping
+++ b/src/Makefile.housekeeping
@@ -10,8 +10,8 @@
 TAB 			:= $(shell $(PRINTF) '\t')
 ECHO_E_ECHO		:= $(ECHO)
 ECHO_E_ECHO_E		:= $(ECHO) -e
-ECHO_E_BIN_ECHO 	:= /bin/echo
-ECHO_E_BIN_ECHO_E 	:= /bin/echo -e
+ECHO_E_BIN_ECHO 	:= $(ECHO)
+ECHO_E_BIN_ECHO_E 	:= $(ECHO) -e
 ECHO_E_ECHO_TAB		:= $(shell $(ECHO_E_ECHO) '\t' | cat)
 ECHO_E_ECHO_E_TAB	:= $(shell $(ECHO_E_ECHO_E) '\t' | cat)
 ECHO_E_BIN_ECHO_TAB 	:= $(shell $(ECHO_E_BIN_ECHO) '\t')
