diff --git a/src/Makefile.housekeeping b/src/Makefile.housekeeping
index cac6efcf..6115bdff 100644
--- a/src/Makefile.housekeeping
+++ b/src/Makefile.housekeeping
@@ -10,8 +10,8 @@
 TAB 			:= $(shell $(PRINTF) '\t')
 ECHO_E_ECHO		:= $(ECHO)
 ECHO_E_ECHO_E		:= $(ECHO) -e
-ECHO_E_BIN_ECHO 	:= /bin/echo
-ECHO_E_BIN_ECHO_E 	:= /bin/echo -e
+ECHO_E_BIN_ECHO 	:= $(ECHO)
+ECHO_E_BIN_ECHO_E 	:= $(ECHO) -e
 ECHO_E_ECHO_TAB		:= $(shell $(ECHO_E_ECHO) '\t' | cat)
 ECHO_E_ECHO_E_TAB	:= $(shell $(ECHO_E_ECHO_E) '\t' | cat)
 ECHO_E_BIN_ECHO_TAB 	:= $(shell $(ECHO_E_BIN_ECHO) '\t')
@@ -1041,7 +1041,7 @@ CLEANUP		+= $(BIN)/NIC	# Doesn't match the $(BIN)/*.* pattern
 
 # Analyse a target name (e.g. "bin/dfe538--prism2_pci.rom.tmp") and
 # derive the variables:
-# 
+#
 # TGT_ELEMENTS : the elements of the target (e.g. "dfe538 prism2_pci")
 # TGT_PREFIX   : the prefix type (e.g. "pcirom")
 # TGT_DRIVERS  : the driver for each element (e.g. "rtl8139 prism2_pci")
@@ -1278,7 +1278,7 @@ $(BIN)/%.zbin : $(BIN)/%.bin $(BIN)/%.zinfo $(ZBIN)
 # Rules for each media format.  These are generated and placed in an
 # external Makefile fragment.  We could do this via $(eval ...), but
 # that would require make >= 3.80.
-# 
+#
 # Note that there's an alternative way to generate most .rom images:
 # they can be copied from their 'master' ROM image using cp and
 # reprocessed with makerom to add the PCI IDs and ident string.  The
@@ -1287,10 +1287,10 @@ $(BIN)/%.zbin : $(BIN)/%.bin $(BIN)/%.zinfo $(ZBIN)
 #   $(BIN)/dfe538%rom : $(BIN)/rtl8139%rom
 #	cat $< $@
 #	$(FINALISE_rom)
-# 
+#
 # You can derive the ROM/driver relationships using the variables
 # DRIVER_<rom> and/or ROMS_<driver>.
-# 
+#
 # We don't currently do this, because (a) it would require generating
 # yet more Makefile fragments (since you need a rule for each ROM in
 # ROMS), and (b) the linker is so fast that it probably wouldn't make
